File: ex01/Triangulo.py
class Triangulo():
    def __init__(self, a, b, c):
        try:
            self.a = int(a)
            self.b = int(b)        
            self.c = int(c)
        except ValueError:
            raise ValueError("Exceção: os lados dos triângulos devem ser um número inteiro.")

    def __str__(self):
        return f"O triângulo é {self.tipo_triangulo()}, pois possui lados {self.a}, {self.b} e {self.c}"

    def tipo_triangulo(self):

        if not self.eh_valido_intervalo_dos_lados():
            return "Fora do intervalo"

        if self.eh_valido():
            if self.a == self.b == self.c:
                return "Equilátero"
            elif self.a == self.b or self.b == self.c or self.a == self.c:
                return "Isósceles"
            else:
                return "Escaleno"
        else:
            return "Não é um triângulo"
    
    def eh_valido(self):
        return (self.a + self.b > self.c and
                self.b + self.c > self.a and
                self.c + self.a > self.b)
    
    def eh_valido_intervalo_dos_lados(self):
        return ((self.a >= 1 and self.a <= 200) and
                (self.b >= 1 and self.b <= 200) and
                (self.c >= 1 and self.c <= 200))

File: assignments/01-triangle/mateus-batista/triangulo/src/main/java/com/exemplo/Triangulo.java
package com.exemplo;

public class Triangulo {
    public static String tipoTriangulo(int a, int b, int c) {
        if (a <= 0 || b <= 0 || c <= 0 || a > 200 || b > 200 || c > 200) return "Lados inválidos";
        if (!(a + b > c && a + c > b && b + c > a)) return "Não é um triângulo";
        if (a == b && b == c) return "Equilátero";
        if (a == b || a == c || b == c) return "Isósceles";
        return "Escaleno";
    }
}
